### ISSUE TITLE
SDL2.dll from VC release doesn't run on macOS Wine

### ISSUE THREAD

**Issue by ericoporto** [libsdl-org/SDL#6996]
- [SDL2-2.24.2.zip](https://github.com/libsdl-org/SDL/releases/download/release-2.24.2/SDL2-2.24.2.zip)
- [SDL2-devel-2.24.2-VC.zip](https://github.com/libsdl-org/SDL/releases/download/release-2.24.2/SDL2-devel-2.24.2-VC.zip)

Here is a weird issue, in AGS, attempt at running any game, on macOS under Wine, using the SDL2.dll that comes with the SDL2-devel-2.24.2-VC release will result in a crash. But if I build it by downloading the source files and using the solution under the VisualC directory to build (msvc2015) it runs fine. We haven't had issues in any other, scenarios with the SDL2.dll that comes from the devel-2.24.2-VC release.

I wanted to build that package locally from source, but turns out the headers in there doesn't match the headers in the source files.

How can I reproduce and build the devel-2.24.2-VC package locally? Is there something that I need to run to generate the headers?

Ideally I would build SDL2.dll from source in our CI so it can run even under Wine. Thanks.

**Comment by slouken** [comment]
That DLL is built using mingw64. Are you able to see what functions are in the stack trace?

**Comment by ericoporto** [comment]
Sorry, I did not thought I would get an answer so soon, I don't have right now but I can do on the weekend.

[GameWithCrashDump.zip](https://github.com/libsdl-org/SDL/files/10348940/GameWithCrashDump.zip)
When I followed this Crash Dump using AGS debug information, I managed to track the issue was SDL_Init on this line here
https://github.com/adventuregamestudio/ags/blob/v.3.6.0.38/Engine/platform/base/sys_main.cpp#L42

But I am not sure yet how to drill down inside of the SDL2 calls.

**Comment by slouken** [comment]
We've been focusing on SDL3, so I'm removing this from the milestone. Please let us know if you have any more information or a fix.

**Comment by ericoporto** [comment]
It would be nice if these releases could be redone shipping the correct headers.

**Comment by madebr** [comment]
[Yesterday's patch release](https://github.com/libsdl-org/SDL/releases/tag/release-2.26.4) looks the same.
I think @slouken needs to update his release script to:
- remove all `SDL_config*` files from the include folder, after that copy and rename `SDL_config_windows.h` to `SDL_config.h` (this is already done) 
- copy `SDL_copying.h` as well, or look into why the VC SDK contains an outdated one.

**Comment by ericoporto** [comment]
I think the best would be to have the releases be made in a CI system instead of something external (perhaps this could be it's own issue), because as is, I can't actually push a fix.

**Comment by slouken** [comment]
Actually SDL_copying.h shouldn't be in the release headers. I'll remove that. The other headers should be correct.

As for the original bug, if I provide a mingw compiled debug DLL, would that help?

**Comment by ericoporto** [comment]
> As for the original bug, if I provide a mingw compiled debug DLL, would that help?

I believe this would make the crash dump have a bit more information on SDL, as it now only knows about our engine. Beyond this, I am not sure exactly what to do with these on macOS..

**Comment by slouken** [comment]
Yeah, I'm doing a fresh copy of the headers in future releases, so this should be fine. None of those headers besides SDL_config.h are used, so that's not related to the crash.

Here is a debug DLL for testing:
[SDL2-macos-wine-debug.zip](https://github.com/libsdl-org/SDL/files/10912845/SDL2-macos-wine-debug.zip)

**Comment by ericoporto** [comment]
I used [cv2pdb](https://github.com/rainers/cv2pdb) to convert the SDL2 you sent, and then got the crash dump [CrashInfo.3.6.0.27.dmp.zip](https://github.com/libsdl-org/SDL/files/10915081/CrashInfo.3.6.0.27.dmp.zip), but I could not yet make neither VS or WinDbg give me a stack of what is happening inside SDL2 still.

So I will probably just switch for building SDL2 from source in our project.

**Comment by ericoporto** [comment]
@slouken , I sent you an email.

**Comment by slouken** [comment]
Hmm, I never got your e-mail. Can you send it to slouken@gmail.com? I'll delete this comment once I get it.

**Comment by ericoporto** [comment]
I sent an email now to this other one. Very short, in case something else I said before triggered some antispam system.

**Comment by ericoporto** [comment]
OK, so turns out I could not reproduce the issue that the other person (who gave me the dumps and the reports) did. I got no crash.

I installed wine using `brew install --cask --no-quarantine wine-stable`, on my macOS Monterrey (so 64-bit only). After enabling sdl2.dll to be run (since it was downloaded from internet) and an AGS windows game. So, AGS is 32-bit and the current recently released wine8 can run 32-bit Windows binaries. So I went and run `wine64 agsgame.exe` and it all worked out. Tried both SDL2.dll built from mingw (default) and my own built using MSVC. Both worked fine.

I have messaged the person who was having the crashes and will update in case I have new information. I will close the issue for the time being.
